code/modules/overmap/overmap_object.dm

```diff
@@ -8,8 +8,22 @@
 
 	var/requires_contact = FALSE //whether or not the effect must be identified by ship sensors before being seen.
 	var/instant_contact  = FALSE //do we instantly identify ourselves to any ship in sensors range?
+	var/halted = FALSE
 	var/sensor_visibility = 10	 //how likely it is to increase identification process each scan.
 
+	var/vessel_mass = 10000             // metric tonnes, very rough number, affects acceleration provided by engines
+	var/vessel_size = SHIP_SIZE_LARGE	// arbitrary number, affects how likely are we to evade meteors
+	var/max_speed = 1/(1 SECOND)        // "speed of light" for the ship, in turfs/tick.
+	var/min_speed = 1/(2 MINUTES)       // Below this, we round speed to 0 to avoid math errors.
+
+	var/list/speed = list(0,0)          // speed in x,y direction
+	var/list/position = list(0,0)       // position within a tile.
+	var/last_burn = 0                   // worldtime when ship last acceleated
+	var/burn_delay = 1 SECOND           // how often ship can do burns
+
+	var/datum/extension/overmap_movement/movement //Movement handler.
```

Comment:
Would suggest setting this to the path and setting extension `if(ispath(movement, /datum/extension))` in `Initialize()`. `movement_handler_type` is otherwise unused, and the naming suggests it is a movement handler, which it isn't.